toolkit/tools/graphpkgfetcher/graphpkgfetcher.go

```diff
@@ -185,13 +313,206 @@ func resolveGraphNodes(dependencyGraph *pkggraph.PkgGraph, inputSummaryFile, out
+		// will pull all of the possible matches). Since we are already matching against SRPMs, we can safely skip these
+		// nodes since they will be included in another node with the same SRPM.
+		if n.Implicit {
+			logger.Log.Debugf("Skipping implicit delta build node %s", n)
+			skippedNodes = append(skippedNodes, n)
+			continue
+		}
+
+		logger.Log.Debugf("Resolving build node %s", n)
+		if n.State == pkggraph.StateBuild {
+			foundMatch, err := downloadSingleDeltaRPM(realDependencyGraph, n, cloner, *outDir, &originalPathToDeltaPathMap)
+			if err != nil {
+				return fmt.Errorf("failed to download delta RPM for build node %s: %w", n, err)
+			}
+			if !foundMatch {
+				// Throw any nodes we fail to resolve the fist time into a list so we can try to resolve them again later.
+				// This will help with cases were a new sub-package is added to a build node, but we won't be able to download
+				// the delta RPM since it doesn't exist yet.
+				skippedNodes = append(skippedNodes, n)
+			}
+		}
+	}
 
-	if strings.TrimSpace(outputSummaryFile) != "" {
-		err = repoutils.SaveClonedRepoContents(cloner, outputSummaryFile)
+	// Fix up the implicit nodes to point to the correct delta RPMs that we parsed earlier.
+	for _, n := range skippedNodes {
+		logger.Log.Debugf("Fixing up skipped node %s", n)
+		err = fixupDeltaImplicitNode(realDependencyGraph, n, &originalPathToDeltaPathMap)
 		if err != nil {
-			logger.Log.Errorf("Failed to save cloned repo contents.")
-			return
+			return fmt.Errorf("failed to fixup skipped node %s: %w", n, err)
+		}
+	}
+
+	logger.Log.Info("Configuring additional delta RPMs as a local repository")
+	err = cloner.ConvertDownloadedPackagesIntoRepo()
+	if err != nil {
+		logger.Log.Errorf("Failed to convert downloaded RPMs into a repo. Error: %s", err)
+		return
+	}
+
+	return
+}
+
+// downloadSingleDeltaRPM attempts to download a single delta RPM for a build node. If the delta RPM is available
+// it will be downloaded and the build node will be updated to point to the new RPM. The associated run node will
+// also be updated to point to the new RPM since the scheduler uses the run node to find the RPM to install.
+//   - realDependencyGraph: The graph to update
+//   - realBuildNode: The build node to update. This node should be from the real graph as we will be updating it directly.
+//     to find the actual build node in the graph.
+//   - cloner: The cloner to use to download the RPMs
+//   - deltaRpmDir: The directory to download the RPMs into (likely the same as the normal RPM cache)
+//   - pathMap: A map of the original path to the delta path for each RPM that was downloaded. This is used to fix up
+//     the implicit nodes later.
+func downloadSingleDeltaRPM(realDependencyGraph *pkggraph.PkgGraph, realBuildNode *pkggraph.PkgNode, cloner *rpmrepocloner.RpmRepoCloner, deltaRpmDir string, pathMap *map[string]string) (foundMatch bool, err error) {
```

Comment:
>*map[string]string [](http://example.com/codeflow?start=166&length=18)

Maps are already passed as refernces, so you can go straight to `map[string]string` and simplify the code a bit.

See: [here](https://stackoverflow.com/questions/40680981/are-maps-passed-by-value-or-by-reference-in-go).

Reply:
Never remember which ones can be assumed to be references... anyways this whole functionality isn't needed for fixing the implicit nodes anymore so the map is gone.